docs/modules/ROOT/pages/references/parameters.adoc

```diff
@@ -6,14 +6,273 @@ The parent key for all of the following parameters is `openshift4_local_storage`
+
+The content of the key is expected to be a dict, with keys in the dict corresponding to labels on namespaces.
+
+The values of the dict can be
+
+* Empty dicts (`{}`).
+In this case, the component constructs the following namespace selector `matchExpressions` entry:
++
+[source,yaml]
+----
+namespaceSelector:
+  labelSelector:
+    matchExpressions:
+      - key: <KEY> <1>
+        operator: DoesNotExist <2>
+----
+<1> The key in the `restricted_to` dict is used as value for field `key` in the match expression
+<2> The `ResourceQuota` needs to be present in all namespaces which don't have the label `<KEY>`.
+
+* Dicts with key `values`.
+In this case, the component constructs the following namespace selector `matchExpressions` entry:
++
+[source,yaml]
+----
+namespaceSelector:
+  labelSelector:
+    matchExpressions:
+      - key: <KEY> <1>
+        operator: NotIn <2>
+        values: < restricted_to[KEY].values > <3>
+----
+<1> The key in the `restricted_to` dict is used as value for field `key` in the match expression
+<2> The `ResourceQuota` needs to be present in all namespaces which don't have the label `<KEY>`.
+<3> The contents of field `values`.
+
+
+== Example configuration
+
+This example configuration shows how to present device `/dev/vdb` on all nodes with label `node-role.kubernetes.io/storage` as a PV with `volumeMode=Block` and storage class `localblock-storage`.
+
+The example restricts the use of the resulting storage class `localblock-storage` to namespaces labelled with `argocd.argoproj.io/instance=rook-ceph`.
+
+[source,yaml]
+----
+parameters:
+  openshift4_local_storage:
+    local_volumes:
+      # Create a `LocalVolume` resource named `storagevolumes`
+      storagevolumes:
+        # Restrict usage of the resulting storage class to namespaces
+        # labelled with `argocd.argoproj.io/instance=rook-ceph`.
+        restricted_to:
+          argocd.argoproj.io/instance:
+            values:
+              - rook-ceph
+        # Present `/dev/vdb` on nodes selected by the node selector
+        # (see below) as PV with `volumeMode=Block` and storage class
+        # `localblock-storage`
+        storage_class_devices:
+          localblock-storage:
+            volumeMode: Block
+            devicePaths:
+              - /dev/vdb
+        config:
+          # Ensure the resulting manager pods can run on nodes tainted
+          # with `storagenode=True:NoSchedule`
+          tolerations:
+            - key: storagenode
+              operator: Exists
+          # Restrict resulting pods to nodes with label
+          # `node-role.kubernetes.io/storage`
+          nodeSelector:
+            nodeSelectorTerms:
+              matchExpressions:
+                - key: node-role.kubernetes.io/storage
+                  operator: Exists
+----
+
+This configuration for parameter `local_volumes` results in the following resources to apply to the cluster:
```

Comment:
I have my doubts here. I see a high risk for the yaml below drifting from reality.

Reply:
Removed YAMLs and replaced them with a high-level description of the resulting objects